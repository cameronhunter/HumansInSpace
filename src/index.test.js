import Skill from './index';
import Request from 'alexa-request';

jest.mock('./space');

test('Launch intent', () => {
  const event = Request.launchRequest().build();
  return expect(Skill(event)).resolves.toMatchSnapshot();
});

test('NumberOfAstronauts intent', () => {
  const event = Request.intent('NumberOfAstronauts').build();
  return expect(Skill(event)).resolves.toMatchSnapshot();
});

test('Yes intent', () => {
  const crafts = { ISS: ['Peggy Whitson', 'Fyodor Yurchikhin', 'Jack Fischer'] };
  const event = Request.intent('AMAZON.YesIntent').session({ attributes: { crafts } }).build();
  return expect(Skill(event)).resolves.toMatchSnapshot();
});

test('Help intent', () => {
  const event = Request.intent('AMAZON.HelpIntent').build();
  return expect(Skill(event)).resolves.toMatchSnapshot();
});

test('Stop intent', () => {
  const event = Request.intent('AMAZON.StopIntent').build();
  return expect(Skill(event)).resolves.toMatchSnapshot();
});

test('Cancel intent', () => {
  const event = Request.intent('AMAZON.CancelIntent').build();
  return expect(Skill(event)).resolves.toMatchSnapshot();
});
