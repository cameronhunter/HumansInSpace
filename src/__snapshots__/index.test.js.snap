// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cancel intent 1`] = `
Object {
  "response": Object {
    "outputSpeech": Object {
      "text": "Live long and prosper! Goodbye.",
      "type": "PlainText",
    },
    "shouldEndSession": true,
  },
  "version": "1.0",
}
`;

exports[`Help intent 1`] = `
Object {
  "response": Object {
    "outputSpeech": Object {
      "text": "I tell you about astronauts currently in space. Would you like to know how many are in orbit?",
      "type": "PlainText",
    },
    "reprompt": Object {
      "outputSpeech": Object {
        "text": "Would you like to know how many astronauts are currently in orbit?",
        "type": "PlainText",
      },
    },
    "shouldEndSession": false,
  },
  "sessionAttributes": Object {
    "crafts": null,
  },
  "version": "1.0",
}
`;

exports[`Launch intent 1`] = `
Object {
  "response": Object {
    "outputSpeech": Object {
      "ssml": "<speak><s>Space, the final frontier!</s><s>There is currently one manned space craft in orbit, with 3 astronauts in total.</s><s>Would you like to hear the crew rosters?</s></speak>",
      "type": "SSML",
    },
    "reprompt": Object {
      "outputSpeech": Object {
        "text": "Would you like me to tell you who is in space?",
        "type": "PlainText",
      },
    },
    "shouldEndSession": false,
  },
  "sessionAttributes": Object {
    "crafts": Object {
      "ISS": Array [
        "Peggy Whitson",
        "Fyodor Yurchikhin",
        "Jack Fischer",
      ],
    },
  },
  "version": "1.0",
}
`;

exports[`NumberOfAstronauts intent 1`] = `
Object {
  "response": Object {
    "outputSpeech": Object {
      "ssml": "<speak><s>Space, the final frontier!</s><s>There is currently one manned space craft in orbit, with 3 astronauts in total.</s><s>Would you like to hear the crew rosters?</s></speak>",
      "type": "SSML",
    },
    "reprompt": Object {
      "outputSpeech": Object {
        "text": "Would you like me to tell you who is in space?",
        "type": "PlainText",
      },
    },
    "shouldEndSession": false,
  },
  "sessionAttributes": Object {
    "crafts": Object {
      "ISS": Array [
        "Peggy Whitson",
        "Fyodor Yurchikhin",
        "Jack Fischer",
      ],
    },
  },
  "version": "1.0",
}
`;

exports[`Stop intent 1`] = `
Object {
  "response": Object {
    "outputSpeech": Object {
      "text": "Live long and prosper! Goodbye.",
      "type": "PlainText",
    },
    "shouldEndSession": true,
  },
  "version": "1.0",
}
`;

exports[`Yes intent 1`] = `
Object {
  "response": Object {
    "outputSpeech": Object {
      "ssml": "<speak><p>On the crew roster for \\"ISS\\", is:</p><s>Peggy Whitson</s><s>Fyodor Yurchikhin</s><s>Jack Fischer</s><break time=\\"1s\\"/><s>Live long and prosper! Goodbye.</s></speak>",
      "type": "SSML",
    },
    "shouldEndSession": true,
  },
  "sessionAttributes": Object {
    "crafts": Object {
      "ISS": Array [
        "Peggy Whitson",
        "Fyodor Yurchikhin",
        "Jack Fischer",
      ],
    },
  },
  "version": "1.0",
}
`;
